tools/SourceKit/tools/sourcekitd-test/sourcekitd-test.cpp

```diff
@@ -1362,6 +1368,12 @@ static void printNameTranslationInfo(sourcekitd_variant_t Info,
 
 static void printCursorInfo(sourcekitd_variant_t Info, StringRef FilenameIn,
                             llvm::raw_ostream &OS) {
+  const char *InternalDiagnostic =
+      sourcekitd_variant_dictionary_get_string(Info, KeyInternalDiagnostic);
+  if (InternalDiagnostic) {
+    OS << "<diagnostic \"" << InternalDiagnostic << "\">\n";
+    return;
```

Comment:
To me, this makes it look like a (true) failure instead of just an empty result.  I suggest something like

```
<empty cursor info>
<internal diagnostic "...">
```

Or 
```
<empty cursor info; internal diagnostic: "...">
```